### ISSUE TITLE
Bug: Forbid invalid `extend` in `interface`

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#11088]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Playground Link

https://typescript-eslint.io/play#ts=5.8.2&fileType=.tsx&code=KYDwDg9gTgLgBASwHY2FAZgQwMbDgMQGcBZCbAazlFSQBNC4AKGATzGAnTnUIEo4A3gFgAUAF8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
export interface FsMock extends (typeof fs) {
}
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

An error with message

> An interface can only extend an identifier/qualified-name with optional type arguments.

### Actual Result

Successfuly parsered.

### Additional Info

All cases mentioned in https://github.com/microsoft/TypeScript/issues/31843 should be rejected.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.30.1` |

**Comment by bradzacher** [comment]
Very similar to https://github.com/typescript-eslint/typescript-eslint/issues/11077

Might be worth adding validation to interface extends?
